src/v3/src/components/PasswordWithConfirmation/PasswordWithConfirmation.tsx

```diff
@@ -0,0 +1,169 @@
+
+enum PasswordType {
+  NEW,
+  CONFIRM,
+}
+
+const PasswordWithConfirmation: UISchemaElementComponent<{
+  uischema: PasswordWithConfirmationElement
+}> = ({ uischema }) => {
+  const { input } = uischema.options;
+  const {
+    name: newPwName,
+    label: newPwLabel,
+    options: {
+      inputMeta: {
+        // @ts-ignore expose type from auth-js
+        messages = {},
+      },
+      attributes,
+    },
+  } = input;
+
+  const { dataSchemaRef } = useWidgetContext();
+  const value = useValue(input);
+  const onChangeHandler = useOnChange(input);
+  // Must use this flag to determine which field contains error
+  const [hasNewPwError, setHasNewPwError] = useState<boolean>(false);
+  const [isNewPwTouched, setIsNewPwTouched] = useState<boolean>(false);
+  const [isTouched, setIsTouched] = useState<boolean>(false);
+  const [confirmPassword, setConfirmPassword] = useState<string>();
+  const newPasswordError = hasNewPwError && messages?.value?.[0]
+    && getMessage(messages.value[0]);
+  const confirmPasswordError = !hasNewPwError
+    ? (messages?.value?.[0] && getMessage(messages.value[0]))
+    : (messages?.value?.[1] && getMessage(messages.value[1]));
+
+  // Overrides default validate function for password field
+  const validate = useCallback((data: FormBag['data']) => {
+    const newPw = data[input.name];
+    const errorMessages: Partial<IdxMessage>[] = [];
+    if (!newPw) {
+      setIsNewPwTouched(false);
+      setHasNewPwError(true);
+      errorMessages.push({ i18n: { key: 'model.validation.field.blank' } });
+    } else {
+      setHasNewPwError(false);
+    }
+
+    if (!confirmPassword) {
+      setIsTouched(false);
+      errorMessages.push({ i18n: { key: 'model.validation.field.blank' } });
+    } else if (confirmPassword !== newPw) {
+      setIsTouched(false);
+      errorMessages.push({ i18n: { key: 'password.error.match' } });
+    }
+
+    return errorMessages.length ? errorMessages : undefined;
+  }, [confirmPassword, input.name, setHasNewPwError, setIsTouched, setIsNewPwTouched]);
+
+  const handlePasswordChange = (e: ChangeEvent<HTMLInputElement>, type: PasswordType) => {
+    if (type === PasswordType.NEW) {
+      setIsNewPwTouched(true);
+      onChangeHandler(e.currentTarget.value);
+    } else {
+      setIsTouched(true);
+      setConfirmPassword(e.currentTarget.value);
+    }
+  };
+
+  const handlePasswordBlur = (type: PasswordType) => {
+    if (type === PasswordType.NEW) {
+      setIsNewPwTouched(true);
+    } else {
+      setIsTouched(true);
+    }
+  };
+
+  useEffect(() => {
+    // update validate function to prevent submission w/o confirm pw data
+    // when server messages are set, we must reset the validate function
+    dataSchemaRef.current![input.name].validate = validate;
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [validate, newPasswordError, confirmPasswordError, input.name]);
+
+  return (
+    <Box>
+      <Box marginBottom={4}>
+        <PasswordInput
```

Comment:
Why current `InputPassword` component cannot be reused? We should be able to just update the uichema to control if rendering field error or not

Reply:
The reason for this is because the current InputPassword component uses its own client-side (onChange/onBlur) validation that conflicts with this custom component. I was using it previously but it was recommended to change it so we can control it more without doing all of the side-stepping previously. 

For example, the current InputPassword component uses the useFieldValidation hook that updates the field error message depending on if it has been changed or not. We need this to be customized for this PasswordWithConfirmation component.